packages/advanced-logic/src/extensions/payment-network/erc777/stream.ts

```diff
@@ -36,64 +36,171 @@ export default class Erc777StreamPaymentNetwork<
   public createCreationAction(
     creationParameters: TCreationParameters,
   ): ExtensionTypes.IAction<TCreationParameters> {
-    if (!creationParameters.expectedFlowRate) {
-      throw Error('expectedFlowRate should not be empty');
+    /* Master Request Creation */
+    if (
+      !(
+        creationParameters.masterRequestId ||
+        creationParameters.previousRequestId ||
+        creationParameters.recurrenceNumber
+      )
```

Comment:
Consider this pattern:
```
const isMasterRequest = condition1;
const isSubsequentRequest = condition2;
if (!isMasterRequest  && !isSubsequentRequest ) {
// throw
}
if (isMaster) {
// ...
}
if (isSubsequent) {
// ...
}
```

Advantages:
- Define both conditions close to each other (they are really similar so easier to check and maintain)
- Exceptions are handled on top as in many similar function
- With the good constant naming, you need less comment

Reply:
Another advantage is to define these 2 conditions somewhere unique, and re-use them in `applyCreationAction`.